lib/WUI/CMakeLists.txt

```diff
@@ -1,29 +1,29 @@
 add_library(WUI INTERFACE)
 
-target_sources(WUI INTERFACE
-  wui.c
-  wui_REST_api.c
-  http/altcp_proxyconnect.c
-  http/fs.c
-  http/httpd.c
-  http/httpd_parser.c
-  lwip.c
-  ethernetif.c
-  netif_settings.c
-  sntp/sntp_client.c
-  sntp/sntp.c
-)
-
-target_include_directories(
+target_sources(
   WUI
-  INTERFACE
-    ${CMAKE_CURRENT_LIST_DIR}
-    http
-    sntp
-    resources
-)
+  INTERFACE wui.c
+            wui_REST_api.c
+            http/altcp_proxyconnect.c
+            http/fs.c
+            http/httpd.c
+            http/httpd_parser.c
+            lwip.c
+            ethernetif.c
+            netif_settings.c
+            sntp/sntp_client.c
+            sntp/sntp.c
+  )
 
-target_link_libraries(
-  WUI
-  INTERFACE LwIP
+add_custom_command(
+  OUTPUT "${PROJECT_ROOT_DIR}/lib/WUI/resources/fsdata_wui_local.c"
+  COMMAND "${Python3_EXECUTABLE}" "${PROJECT_ROOT_DIR}/utils/wui.py"
+  COMMENT "Generating fsdata_wui_local raw data from WUI resources."
+  VERBATIM
   )
+
+add_custom_target(wui-genfile DEPENDS "${PROJECT_ROOT_DIR}/lib/WUI/resources/fsdata_wui_local.c")
```

Comment:
/question/ I might be missing something, but where is this target being used? Is it meant for direct invocation from CLI only? I would expect the WUI library to have it as a dependency.

Reply:
This script is not meant to be used for WUI library, just for client side of FW. WUI library does not need any dependencies for this.